configstack/module.go

```diff
@@ -52,7 +53,68 @@ func ResolveTerraformModules(terragruntConfigPaths []string, terragruntOptions *
 		return []*TerraformModule{}, err
 	}
 
-	return crosslinkDependencies(mergeMaps(modules, externalDependencies), canonicalTerragruntConfigPaths)
+	crossLinkedModules, err := crosslinkDependencies(mergeMaps(modules, externalDependencies), canonicalTerragruntConfigPaths)
+	if err != nil {
+		return []*TerraformModule{}, err
+	}
+
+	finalModules := removeExcludedDirs(crossLinkedModules, terragruntOptions)
```

Comment:
Should modules be excluded *before* cross-linking?

Reply:
I excluded them *after* cross-linking, because the depency information is needed. This information is used to exclude modules with dependencies which are in the to-be-excluded list. (See other conversation)

Reply:
However even if we remove them before cross-linking, I think `cross-linking` will throw an error, because it can no longer find some modules which are specified as dependencies.